backend/fms_core/template_importer/row_handlers/sample_submission/sample.py

```diff
@@ -59,27 +60,41 @@ def process_row_inner(self, sample, library, container, project, parent_containe
                                individual["reference_genome"],
                                mother_obj,
                                father_obj])
+        can_use_generic_individual = (any([taxon_obj is not None,
+                                           reference_genome_obj is not None])
+                                      and not any([individual["pedigree"],
+                                                   mother_obj,
+                                                   father_obj]))
         # When the individual name is not provided any field that is stored on the individual need to raise an error.
         self.errors['individual'] = []
         self.warnings['individual'] = []
-        if not individual["name"] and need_individual:
-            if individual["taxon"]:
-                self.errors['individual'].append(f"Individual taxon requires an individual name to be provided to be saved.")
+        if not individual["name"] and need_individual and not can_use_generic_individual:
```

Comment:
This if condition is kinda hard to wrap my mind around. I think it should be in a variable.